src/main/java/edu/kis/powp/jobs2d/drivers/visitors/SimpleDriverCountingVisitor.java

```diff
@@ -0,0 +1,57 @@
+package edu.kis.powp.jobs2d.drivers.visitors;
+
+import edu.kis.powp.jobs2d.drivers.ComplexDriver;
+import edu.kis.powp.jobs2d.drivers.InformativeLoggerDriver;
+import edu.kis.powp.jobs2d.drivers.VisitableJob2dDriver;
+import edu.kis.powp.jobs2d.drivers.adapter.LineDriverAdapter;
+import edu.kis.powp.jobs2d.drivers.monitoring.DriverMonitorDecorator;
+import edu.kis.powp.jobs2d.transformations.RotateTransformationDecorator;
+import edu.kis.powp.jobs2d.transformations.ScaleTransformationDecorator;
+
+public class SimpleDriverCountingVisitor implements DriverVisitor {
+    private int total;
+
+    @Override
+    public void visit(ComplexDriver driver) {
+        total = 0;
```

Comment:
ComplexDriver can contain ComplexDriver (consider adding such a deep complex driver for testing).
You should use addtional local variable, e.g.
int res = 1; // for current ComplexDriver instance
increment it in the loop res += total
and finally set total = res.